### PR TITLE
Add aria role of option,aria-selected and tabindex based keyboard navigation

### PR DESCRIPTION
**Pull request by mpnkhan**
Refs #13556.

### CODE REVIEW

**Comment by cvrebert** [comment]
CC: @patrickhlauke for review

File: docs/_includes/js/carousel.html [hunk 1]
```diff
@@ -4,20 +4,20 @@ <h1 id="carousel" class="page-header">Carousel <small>carousel.js</small></h1>
   <h2 id="carousel-examples">Examples</h2>
   <p>The slideshow below shows a generic plugin and component for cycling through elements like a carousel.</p>
   <div class="bs-example">
-    <div id="carousel-example-generic" class="carousel slide" data-ride="carousel">
+    <div id="carousel-example-generic" class="carousel slide" data-ride="carousel" data-interval="false" data-wrap="false">
```
Comment (patrickhlauke) [inline docs/_includes/js/carousel.html]: I'd agree that the defaults we give in our examples should be the ones that cause the least amount of problems, and then explain to devs how they can knowingly change them, with a note to make them aware of the consequences

**Comment by mdo** [comment]
Probably worth punting on adding this to v3? No new features and all.

**Comment by patrickhlauke** [comment]
On one hand, I think because it's been sitting here so long (and the fact that our carousel is very badly in need of a bit of accessibility love)  I'd love to merge it as an a11y bugfix. On the other, not sure how much rebasing is involved?

**Comment by cvrebert** [comment]
* I think it's feasible to apply most of this patch 
* I would vote to classify it as a bugfix, the bug being that the carousel is unnecessarily inaccessible.

**Comment by mdo** [comment]
Given the lack of activity here, and the age of the PR, I'm inclined to punt to v4. Thoughts?

**Comment by patrickhlauke** [comment]
Agree

**Pull request closed without merging**